packages/components/src/components/side-nav/side-nav.component.ts

```diff
@@ -0,0 +1,291 @@
+   */
+  @query('slot:not([name])') private defaultSlot: HTMLSlotElement;
+
+  /**
+   * Reference to the footer slot.
+   */
+  @query('slot[name="footer"]') private footerSlot: HTMLSlotElement;
+
+  /**
+   * Reference to the drawer
+   */
+  @query('.side-nav__drawer') private drawer: SynDrawer;
+
+  /**
+   * State if all nav-items have a prefix icon.
+   */
+  @state() private hasPrefixIcons = false;
+
+  /**
+   * Indicates whether or not the side-nav is open.
+   * You can toggle this attribute to show and hide the side-nav, or you can use the `show()` and
+   * `hide()` methods and this attribute will reflect the side-nav's open state.
+   *
+   * Depending if the rail attribute is set or not, the behavior will differ.
+   *
+   * __Non rail__:
+   * With `open` will show the side-nav.
+   * Without `open`, the side-nav will be hidden.
+   *
+   * __Rail__:
+   * With `open` will show the whole side-nav with an overlay for touch devices
+   * or without an overlay for non-touch devices.
+   * Without `open`, the side-nav will only show the prefix of nav-item's.
+   *
+   */
+  @property({ reflect: true, type: Boolean }) open = false;
+
+  /**
+   * Use the rail attribute to only show the prefix of navigation items in closed state.
+   * This will open on hover on the rail navigation.
+   * On touch devices the navigation opens on click and shows an overlay.
+   *
+   * Note: The Rail is only an option if all Navigation Items on the first level have an Icon.
+   * If this is not the case you should use a burger navigation.
+   */
+  @property({ reflect: true, type: Boolean }) rail = false;
+
+  /**
+   * Get all nav-items from the default and footer slot.
+   */
+  private getAllNavItems(): SynNavItem[] {
+    const navItemsAll: SynNavItem[] = [];
+
+    const addNavItems = (element: Element) => {
+      if (element.tagName.toLowerCase() === 'syn-nav-item') {
+        navItemsAll.push(element as SynNavItem);
+      } else {
+        // Check for first child nav-items.
+        // It`s possible that they were wrapped in another element (e.g. div or nav)
+        const nestedNavItems = Array.from(element.children).filter(nestedElement => (nestedElement.tagName.toLowerCase() === 'syn-nav-item')) as SynNavItem[];
+        navItemsAll.push(...nestedNavItems);
+      }
+    };
+
+    (this.defaultSlot?.assignedElements({ flatten: true }) || []).forEach(addNavItems);
+    (this.footerSlot?.assignedElements({ flatten: true }) || []).forEach(addNavItems);
+
+    return navItemsAll;
+  }
+
+  /**
+   * This function handles the rail mode.
+   * Rail mode is only valid if every nav-item has a prefix.
+   */
+  // eslint-disable-next-line complexity
+  private handleRailMode() {
+    const navItems = this.getAllNavItems();
+
+    const itemsWithPrefix = navItems.filter((navItem) => !!navItem.querySelector(':scope > [slot="prefix"]'));
+
+    this.hasPrefixIcons = navItems.length !== 0 && itemsWithPrefix.length === navItems.length;
```

Comment:
I'm curious: What would happen if someone doesn't set a prefix and we wouldn't have this check?

Reply:
And if we really need this: How can we indicate users, that they have to set the prefixes? Console.log?